apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -98,31 +98,42 @@ export const updatePageById = (
   page: Partial<Omit<Page, "id">> & { id: number },
 ) => {
   const { id, ...rest } = page
-  return db.transaction().execute((tx) => {
-    return tx
-      .updateTable("Resource")
-      .set(rest)
-      .where("id", "=", id)
-      .executeTakeFirstOrThrow()
-  })
+  return db
+    .updateTable("Resource")
+    .set(rest)
+    .where("id", "=", id)
+    .executeTakeFirstOrThrow()
 }
 
-export const updateBlobById = (props: {
+// NOTE: This id here is the page id
+export const updateBlobById = async (props: {
   id: number
   content: PrismaJson.BlobJsonContent
 }) => {
   const { id, content } = props
-  return (
-    db
-      .updateTable("Blob")
-      .innerJoin("Resource", "Resource.id", "id")
-      // NOTE: This works because a page has a 1-1 relation with a blob
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore excessive deep type instantiaton
-      .set({ content })
+  return db.transaction().execute(async (tx) => {
+    const page = await tx
+      .selectFrom("Resource")
       .where("Resource.id", "=", id)
+      // NOTE: We update the draft first
+      // Main should only be updated at build
+      .select("draftBlobId")
       .executeTakeFirstOrThrow()
-  )
+
+    if (!page.draftBlobId) {
+      // NOTE: no draft for this yet, need to create a new one
+      return tx.insertInto("Blob").values({ content }).executeTakeFirst()
```

Comment:
after creating draftBlob, we need to update resource's draftBlobId to this newly created id right?

Reply:
good catch, updated